1. Basics/task_2/hw2.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * @return string the input request example as a string
+ */
+function readHttpLikeInput() : string {
+    $f = fopen( 'php://stdin', 'r' );
+    $store = "";
+    $toread = 0;
+    while( $line = fgets( $f ) ) {
+        $store .= preg_replace("/\r/", "", $line);
+
+        // store a request body length if the body exsist
+        // $m - array of matches
+        // $m[1] <- (\d+); $m[1]*1 - (str->int) - the length of the body
+        if (preg_match('/Content-Length: (\d+)/',$line,$m)) {
+            $toread=$m[1]*1;
+        }
+
+        if ($line == "\r\n") 
+              break;
+    }
+
+    // if the request has a body, read and store it
+    if ($toread > 0) 
+        $store .= fread($f, $toread);
+    
+    return $store;
+}
+
+$contents = readHttpLikeInput();
+
+/**
+ * @param string $string : input request sample as a string
+ *
+ * @return array of the needed request parts
+ */
+function parseTcpStringAsHttpRequest($string) : array {
+    $strAsArr = explode("\n", $string);
+    $methodAndURI = explode(" ", array_shift($strAsArr));
+    $headers = [];
+    $body = NULL;
+
+    // if the request has headers store each header data as
+    // $headers[headerName] = headerValue
+    if($strAsArr) {
+        $i = 0;
+        while($strAsArr[$i]) {
+            $ind = strpos($strAsArr[$i], ":");
```

Comment:
try to simplify this logic
there is a simpler way to do it.

Reply:
Done.